tests/core/test_fileio.py

```diff
@@ -21,161 +21,116 @@
-            'params': {
-                'feat_in': 64,
-                'activation': 'relu',
-                'conv_mask': True,
-                'jasper': [
-                    {
-                        'filters': 1024,
-                        'repeat': 1,
-                        'kernel': [1],
-                        'stride': [1],
-                        'dilation': [1],
-                        'dropout': 0.0,
-                        'residual': False,
-                        'separable': True,
-                        'se': True,
-                        'se_context_size': -1,
-                    }
-                ],
-            },
-        }
-
-        decoder = {
-            'cls': 'nemo.collections.asr.modules.ConvASRDecoder',
-            'params': {
-                'feat_in': 1024,
-                'num_classes': 28,
-                'vocabulary': [
-                    ' ',
-                    'a',
-                    'b',
-                    'c',
-                    'd',
-                    'e',
-                    'f',
-                    'g',
-                    'h',
-                    'i',
-                    'j',
-                    'k',
-                    'l',
-                    'm',
-                    'n',
-                    'o',
-                    'p',
-                    'q',
-                    'r',
-                    's',
-                    't',
-                    'u',
-                    'v',
-                    'w',
-                    'x',
-                    'y',
-                    'z',
-                    "'",
-                ],
-            },
-        }
-        modelConfig = DictConfig(
-            {'preprocessor': DictConfig(preprocessor), 'encoder': DictConfig(encoder), 'decoder': DictConfig(decoder)}
-        )
-        asr_model = EncDecCTCModel(cfg=modelConfig)
+    def test_save_restore_from_nemo_file(self, asr_model):
+        """" Test makes sure that the second instance created from the same configuration AND checkpoint 
+        has the same weights. """
 
         with tempfile.NamedTemporaryFile() as fp:
             filename = fp.name
             asr_model.save_to(save_path=filename)
+
+            # Restore!
             asr_model2 = EncDecCTCModel.restore_from(restore_path=filename)
-            self.assertEqual(len(asr_model.decoder.vocabulary), len(asr_model2.decoder.vocabulary))
-            self.assertEqual(asr_model.num_weights, asr_model2.num_weights)
+
+            assert len(asr_model.decoder.vocabulary) == len(asr_model2.decoder.vocabulary)
+            assert asr_model.num_weights == asr_model2.num_weights
+
             w1 = asr_model.encoder.encoder[0].mconv[0].conv.weight.data.detach().cpu().numpy()
             w2 = asr_model2.encoder.encoder[0].mconv[0].conv.weight.data.detach().cpu().numpy()
-            self.assertTrue(np.array_equal(w1, w2))
+
+            print(w1)
+            print(w2)
```

Comment:
Ech, missed those two. Still, pytest won't print them (unless an error will pop out).

Will fix this in next PR - where I will try to unify the serialization/checkpointing for modules/models with pure cfg and list of arguments.

Reply:
fixed